easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialog.java

```diff
@@ -185,8 +262,21 @@ public Builder setRequestCode(int requestCode) {
          * call to {@link AppSettingsDialog#show()}.
          */
         public AppSettingsDialog build() {
-            return new AppSettingsDialog(mActivityOrFragment, mContext, mRationale, mTitle,
-                    mPositiveButton, mNegativeButton, mNegativeListener, mRequestCode);
+            mPositiveButton = TextUtils.isEmpty(mPositiveButton) ?
+                    mContext.getString(android.R.string.ok) : mPositiveButton;
+            mNegativeButton = TextUtils.isEmpty(mNegativeButton) ?
+                    mContext.getString(android.R.string.cancel) : mNegativeButton;
+            mRequestCode = mRequestCode > 0 ? mRequestCode : DEFAULT_SETTINGS_REQ_CODE;
```

Comment:
I'm pretty sure the convention is to set up your variables in the build method. From effective java, the `build` method sets stuff up and the constructor just expects to receive the right stuff.